firestore/google/cloud/firestore_v1/client.py

```diff
@@ -112,8 +114,23 @@ def _firestore_api(self):
             <The GAPIC client with the credentials of the current client.
         """
         if self._firestore_api_internal is None:
+            SERVICE_ADDRESS = "firestore.googleapis.com:443"
+
+            # Use a custom channel.
+            # We need this in order to set appropriate keepalive options.
+            channel = grpc_helpers.create_channel(
+                target=SERVICE_ADDRESS,
```

Comment:
This should probably follow the pattern in pubsub: https://github.com/googleapis/google-cloud-python/blob/b80d431e214c62fb506461a3f1485b58bf38f289/pubsub/google/cloud/pubsub_v1/publisher/client.py#L74-L96

Reply:
I used PubSub for inspiration, but didn't want to extend the client further. Do you have a specific way in which this is different that you disagree with?

Reply:
A couple:
- DRY the service address (using `self.target`).
- Passing in a transport to the client is how the GAPIC-generator folks want us to customize such stuff, but doing it here (rather than in `__init__` means we can't let the caller pass it in to override our choices).

Reply:
@tseaver, as this is about addressing a bug, I wasn't intending to modify the user surface. the `self.target` callout is a good idea though.

Also, I opened https://github.com/googleapis/gapic-generator/issues/2824 to see about getting better exposure the options in the generator, which would minimize the change needed here to specify grpc options.

Reply:
@crwilcox Did you mean to push a change which used the `self.target` DRYness?